plugins/connections/ui/src/main/java/org/pentaho/di/connections/ui/tree/ConnectionFolderProvider.java

```diff
@@ -58,4 +60,14 @@ public void refresh( AbstractMeta meta, TreeNode treeNode, String filter ) {
   public String getTitle() {
     return STRING_VFS_CONNECTIONS;
   }
+
+  @Override
+  public void create( AbstractMeta meta, TreeNode parent ) {
+    Repository repository = Spoon.getInstance().getRepository();
```

Comment:
You'll probably get a check style for this `repository!=null `--> `repository != null`